lib/model/store.dart

```diff
@@ -1142,30 +1130,40 @@ class UpdateMachine {
         shouldReportToUser = true;
 
       case ZulipApiException(code: 'BAD_EVENT_QUEUE_ID'):
-        return false;
+        Error.throwWithStackTrace(error, stackTrace);
 
       case ZulipApiException():
       case MalformedServerResponseException():
         // Either a 4xx we didn't expect, or a malformed response;
         // in either case, a mismatch of the client's expectations to the
         // server's behavior, and therefore a bug in one or the other.
         // TODO(#1054) handle auth failures specifically
-        return false;
+        Error.throwWithStackTrace(error, stackTrace);
     }
 
     assert(debugLog('Transient error polling event queue for $store: $error\n'
         'Backing off, then will retry…'));
     if (shouldReportToUser) {
       _maybeReportToUserTransientError(error);
     }
-    return true;
+    await (_pollBackoffMachine ??= BackoffMachine()).wait();
+    if (_disposed) return;
+    assert(debugLog('… Backoff wait complete, retrying poll.'));
   }
 
-  /// Sort out an error in [poll].
+  /// Deal with an error in [poll]: reload server data to replace the store,
+  /// after reporting the error as appropriate to the user and/or developer.
   ///
-  /// Reports the error if appropriate to the user and/or developer;
-  /// then returns true just if the error was unexpected.
-  bool _triagePollError(Object error) {
+  /// See also:
+  ///  * [_handlePollRequestError], which handles certain errors
+  ///    and causes them not to reach this method.
+  Future<void> _handlePollError(Object error) async {
```

Comment:
Since this helper becomes the only user of `_unexpectedErrorBackoffMachine`, would it be better to move that `BackoffMachine` here?

Reply:
Hmm, good thought.

I'll actually move it next to `_pollBackoffMachine` (which is also moving in this commit — so in fact this basically means they stay together). I think it's useful context when looking at one backoff machine to see that there's also this other related one.